src/main/java/io/zeebe/kafka/connect/source/ZeebeSourceTask.java

```diff
@@ -68,39 +68,38 @@ public void start(final Map<String, String> props) {
     workers =
         jobTypes.stream().map(type -> this.newWorker(config, type)).collect(Collectors.toList());
 
-    running.set(true);
+    if (!taskState.compareAndSet(TaskState.STOPPED, TaskState.STARTED)) {
```

Comment:
I'm not sure whether this state handling is necessary or not

Reply:
I couldn't find anything on how often start/stop is called in the lifetime of a task, so I figured a fail-safe could be useful. It's probably safe to assume that start is only called if the task is new or it was previously stopped before, but I figured just in case (plus it's not the hot path so a little performance hit there isn't a huge deal I think).